src/components/common/PaginatedTxns/index.tsx

```diff
@@ -9,93 +10,78 @@ import InfiniteScroll from '../InfiniteScroll'
+const getFilterResultCount = (filter: TxFilter, page: TransactionListPage) => {
   const count = page.results.filter(isTransactionListItem).length
 
   return `${page.next ? '> ' : ''}${count} ${filter.type} transactions found`.toLowerCase()
 }
 
-const TxPage = ({
-  pageUrl,
-  useTxns,
-  onNextPage,
-  isFirstPage,
-}: {
-  pageUrl: string
-  useTxns: typeof useTxHistory | typeof useTxQueue
-  onNextPage?: (pageUrl?: string) => void
-  isFirstPage: boolean
-}): ReactElement => {
-  const { page, error } = useTxns(pageUrl)
+const PaginatedTxns = ({ useTxns }: { useTxns: typeof useTxHistory | typeof useTxQueue }): ReactElement => {
+  const [pages, setPages] = useState<Array<{ pageUrl?: string; data: TransactionListPage }>>([])
+  const [pageUrl, setPageUrl] = useState<string>()
   const [filter] = useTxFilter()
-
   const isQueue = useTxns === useTxQueue
+  const allItems = useMemo(() => pages.flatMap((page) => page.data.results), [pages])
 
-  if (page?.results) {
-    return (
-      <>
-        {/* FIXME: batching will only work for the first page results */}
-        {isFirstPage && (
-          <Box display="flex" flexDirection="column" alignItems="flex-end" mt={['-94px', '-44px']} mb={['60px', 0]}>
-            {isQueue ? <BatchExecuteButton items={page.results} /> : <TxFilterButton />}
-            {filter && <Typography mt={2}>{getResultCount(filter, page)}</Typography>}
-          </Box>
-        )}
-
-        {page.results.length ? <TxList items={page.results} /> : isQueue && <NoQueuedTxns />}
-
-        {onNextPage && page.next && (
-          <Box my={4} textAlign="center">
-            <InfiniteScroll onLoadMore={() => onNextPage(page.next)} />
-          </Box>
-        )}
-      </>
-    )
-  }
-
-  if (error) {
-    return <ErrorMessage>Error loading transactions</ErrorMessage>
-  }
-
-  return <SkeletonTxList />
-}
+  // Reset the pages when the filter changes
+  useEffect(() => {
+    setPages([])
+    setPageUrl(undefined)
+  }, [filter, useTxns])
 
-const PaginatedTxns = ({ useTxns }: { useTxns: typeof useTxHistory | typeof useTxQueue }): ReactElement => {
-  const [pages, setPages] = useState<string[]>([''])
-  const [filter] = useTxFilter()
+  // Load the current page
+  const { page: lastPage, error, loading } = useTxns(pageUrl)
 
+  // When a tx page is loaded, update the list of pages
   useEffect(() => {
-    setPages([''])
-  }, [filter])
+    if (!lastPage) return
 
-  const onNextPage = (pageUrl = '') => {
-    setPages((prev) => [...prev, pageUrl])
-  }
+    setPages((prevPages) => {
+      // If we're on the first page), "refresh" that page because it's polled.
+      // Otherwise, append the new page to the list of pages.
+      let pageIndex = prevPages.findIndex((page) => page.pageUrl === pageUrl)
+      if (pageIndex === -1) pageIndex = prevPages.length
+      const newPages = prevPages.slice()
+      newPages[pageIndex] = { pageUrl, data: lastPage }
+      return newPages
+    })
```

Comment:
Was there an issue with the previous approach of always appending the new page? e.g. `setPages((prevPages) => [...prevPages, { pageUrl, data: lastPage }])`

Reply:
Yes, actually the bug you found. If you're on the first page, the page updates every 15 sec and we don't won't to append it every time.

Reply:
I've just realized there might be another bug. When a tx is executed, the first history page updates and there are more txns, then the pageUrl might be also different (a new offset).

Reply:
Good find! I was also able to reproduce that issue. We only update the history if there is no filter applied or the user hasn't scrolled yet. Not sure if we should "force" an update in those cases.

Reply:
Fixed it by using a count.

Reply:
I ended up restoring the previous approach with separate TxPage components each with its own loading hook.